src/Forms/Submission.php

```diff
@@ -52,7 +52,7 @@ public function date($date = null)
             $this->date = $date;
         }
 
-        return $this->date;
+        return $this->date = Carbon::now();
```

Comment:
Would be better as `$this->date ?? ($this->date = Carbon::now())` so that it doesnt overwrite the existing value